### ISSUE TITLE
check_by_ssh skip hostname check [sf#3611614]

### ISSUE THREAD

**Issue by monitoring-user** [monitoring-plugins/monitoring-plugins#1135]
Submitted by francyci on 2013-04-23 01:19:44

Please add a feature in check_by_ssh to skip the host_or_die() check.

I have host aliases defined in my system ssh_config (/etc/ssh/ssh_config) that are only accessible via ProxyCommand.  There are no DNS records for those systems since they exist outside the local network, and are not accessible except through a Proxy.

check_by_ssh.c

```
            case 'H':                                                                       /* host */
                    host_or_die(optarg);
                    hostname = optarg;
                    break;
```

If I comment out the `host_or_die()` and re-compile check_by_ssh seems to work perfectly fine to connect to the systems I have defined with aliases in my ssh_config.

**Comment by RincewindsHat** [comment]
Should check_by_ssh consider such things or stay (relatively) simple?
I could imagine a scenario where a host A is behind a ProxyHost, but A is not a valid DNS name in the network of the monitoring host.
Otherwise I would suggest to close this issue.

**Comment by zoredache** [comment]
Well, I am still running a locally patched version. So I am still hoping there will be an official fix.

**Comment by RincewindsHat** [comment]
closed with #1691
